lib/create-testing-library-rule.ts

```diff
@@ -0,0 +1,37 @@
+import { ESLintUtils, TSESLint } from '@typescript-eslint/experimental-utils';
+import { getDocsUrl } from './utils';
+import {
+  detectTestingLibraryUtils,
+  DetectionHelpers,
+} from './detect-testing-library-utils';
+
+type CreateRuleMetaDocs = Omit<TSESLint.RuleMetaDataDocs, 'url'>;
+type CreateRuleMeta<TMessageIds extends string> = {
+  docs: CreateRuleMetaDocs;
+} & Omit<TSESLint.RuleMetaData<TMessageIds>, 'docs'>;
+
+export function createTestingLibraryRule<
+  TOptions extends readonly unknown[],
+  TMessageIds extends string,
+  TRuleListener extends TSESLint.RuleListener = TSESLint.RuleListener
+>(
+  config: Readonly<{
+    name: string;
+    meta: CreateRuleMeta<TMessageIds>;
+    defaultOptions: Readonly<TOptions>;
+    create: (
+      context: Readonly<TSESLint.RuleContext<TMessageIds, TOptions>>,
+      optionsWithDefault: Readonly<TOptions>,
+      detectionHelpers: Readonly<DetectionHelpers>
+    ) => TRuleListener;
+  }>
+) {
+  const { create, ...remainingConfig } = config;
+
+  return ESLintUtils.RuleCreator(getDocsUrl)({
+    ...remainingConfig,
+    create: detectTestingLibraryUtils<TOptions, TMessageIds, TRuleListener>(
+      create
+    ),
```

Comment:
Where all the magic happens now. It looks **so** simple after all the time I spent on it that I feel kinda stupid.

Reply:
Nice!
It actually looks easier and cleaner than expected ... that's easy to say when it's written.

Reply:
this looks great! Great job!